--- squashfs-tools/mksquashfs.h	2021-01-22 20:48:23.000000000 +0100
+++ squashfs-tools/mksquashfs.h	2021-01-22 20:48:30.000000000 +0100
@@ -144,7 +144,7 @@
 #endif
 
 extern struct cache *reader_buffer, *fragment_buffer, *reserve_cache;
-struct cache *bwriter_buffer, *fwriter_buffer;
+extern struct cache *bwriter_buffer, *fwriter_buffer;
 extern struct queue *to_reader, *to_deflate, *to_writer, *from_writer,
 	*to_frag, *locked_fragment, *to_process_frag;
 extern struct append_file **file_mapping;
